libvast/src/system/catalog.cpp

```diff
@@ -455,16 +492,31 @@ catalog(catalog_actor::stateful_pointer<catalog_state> self,
           = std::unique(ids_candidates.begin(), ids_candidates.end());
         ids_candidates.erase(new_end, ids_candidates.end());
       }
-      std::vector<vast::uuid> result_candidates;
-      if (has_expression && has_ids)
+      std::vector<partition_info> result_candidates;
+      if (has_expression && has_ids) {
         std::set_intersection(expression_candidates.partitions.begin(),
                               expression_candidates.partitions.end(),
                               ids_candidates.begin(), ids_candidates.end(),
                               std::back_inserter(result_candidates));
-      else if (has_expression)
+      } else if (has_expression) {
         result_candidates = std::move(expression_candidates.partitions);
-      else
-        result_candidates = std::move(ids_candidates);
+      } else {
+        result_candidates.reserve(ids_candidates.size());
+        for (const auto& ids_candidate : ids_candidates) {
+          const auto syn = self->state.synopses.find(ids_candidate);
+          if (syn == self->state.synopses.end())
+            VAST_WARN("{} ignores unknown candidate partition hint {} for "
+                      "query {}",
+                      *self, ids_candidate, query.id);
```

Comment:
Can you add a comment about why this check is needed?

Reply:
Turns out it wasn't. I added a comment why and added an assertion instead.